docs/src/ci.md

```diff
@@ -0,0 +1,143 @@
+
+Before the unit tests are executed, the `SetupDevEnv.jl` is executed, which prepares the project environment for the unit test. It reads the `Project.toml` of the current project and adds the version of the `dev` branch of all QED dependency (`Pkg.develop()`) if no line starting with `CI_UNIT_PKG_URL_` was defined in the commit message. If `CI_UNIT_PKG_URL_` was defined, it use a custom URL.
+
+The commit message is defined in the environment variable `CI_COMMIT_MESSAGE` by GitLab CI. If the variable is not defined, the script ignores the commit message. If you want to disable reading the commit message you can set the name of the commit message variable to a undefined variable via first argument of the `SetupDevEnv.jl` script. We use this, when we execute the CI on the `main` or `dev` branch. On this branches, it should be not possible to use custom URLs for unit or integration tests. Therefore we disable it, which also allows to use `CI_UNIT_PKG_URL_` variables as regular part of the merge commit message.
+
+## running locally
+
+If you want to run the script locally, you can set custom URLs via environment variables. For example:
+
+```bash
+CI_UNIT_PKG_URL_QEDbase="www.github.com/User/QEDbase#0e1593b" CI_UNIT_PKG_URL_QEDfields="www.github.com/User/QEDfields#60324ad" julia --project=/path/to/QED/repo SetupDevEnv.jl`
+```
+
+# Integration Tests for CI Develops
+
+In this section, we explain how the integration tests are created and executed. It is not mandatory to read the section if you only want to use the CI. The following figure shows the stages of the CI pipeline:
+
+![detailed CI pipeline of the integration tests](integration_jobs_pipeline.svg)
+
+All of the following stages are executed in the CI pipeline of the sub-package, where the code is modified. This means also, that GitLab CI automatically checks out the repository with the changes of the pull request and provides it in the CI job. We name the package `orig` for easier understanding of the documentation.
+
+- **Stage: Integration Tests**: Execute the unit tests of `orig` via `Pkg.test()`.
+- **Stage: Generate integration Tests**: Download the `integTestGen.jl` script from the `QED.jl` package via `git clone`. The script is executed with the name of the subpackage `orig`. The script traverses the dependency tree of the `QED.jl` and it is searching for sub-packages which has a dependency of package `orig`. For each package with the dependency of package `orig`, the generator generates a job yaml. By default, it uses the upstream repository and development branch of the package tested in the integration test. The repository and commit can be changed via the environment parameter in the git commit message. The `integTestGen.jl` uses the `Package.toml` of the `QED.jl` package because all sub-packages are direct and indirect dependencies of it.
```

Comment:
```suggestion
- **Stage: Generate integration Tests**: Download the `integTestGen.jl` script from the `QED.jl` package via `git clone`. The script is executed with the name of the subpackage `orig`. The script traverses the dependency tree of `QED.jl` and searches for sub-packages which depend on `orig`. For each package that depends on `orig`, the generator generates a job yaml. By default, it uses the upstream repository and development branch of the package tested in the integration test. The repository and commit can be changed via the environment parameter in the git commit message. The `integTestGen.jl` uses the `Package.toml` of the `QED.jl` package because all sub-packages are direct and indirect dependencies of it.
```

It was hard to understand who depends on whom. If I understood it correctly, then the edit should make it more clear.